transcom/mymove
# B 21536 int

## [B-21536](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A1021768)

## Summary

When a TOO approves multiple requested update to a shipments address, created by Prime, the Move History tab was only displaying the most recent approval. This was due to some columns being ignored, and the address being replaced by ID rather than being actually updated. 

Another issue, found by @MInthavongsay, where Secondary addresses boolean being set to 'No' after having previously been 'Yes' was delisting old logs related to secondary addresses was fixed as well. The test steps have been updated to reflect these changes.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

Setup: Create a shipment with an HHG and push it through the process until it is available to prime.

1. Access the Prime Sim
2. Access your create HHG
3. Request to Update Shipment Destination Address (Be sure the new address is in a different service area so it requires TOO approval)
4. Login as a TOO
5. Navigate to the created HHG
6. Select to edit the shipment
7. Select to review the shipment destination address change
8. Approve the shipment destination address change
9. Navigate to the move history tab and validate that the new address update shows on the page.
10. Repeat steps 1-9 again and validate both updates are still present in move history

Test steps for secondary addresses
1. Login as a TOO
2. Navigate to a move with an HHG and add a secondary address.
3. Save the shipment.
4. View the Move History and see that it displayed the added shipment
5. Remove the secondary address from the shipment
6. View the Move History and see that it displays both the addition and removal of the secondary address
7. Repeat steps 2-6 multiple times on the same shipment to verify logs are retained

### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
![Screenshot 2024-12-10 at 11 00 41 AM](https://github.com/user-attachments/assets/5772cec4-f092-48b4-b9ce-8a65c78b33e8)
